activity/src/test/java/com/amazonaws/stepfunctions/cloudformation/activity/DeleteHandlerTest.java

```diff
@@ -80,4 +95,34 @@ public void testUnknownException() {
         assertThat(response.getMessage()).isEqualTo(Constants.INTERNAL_FAILURE_MESSAGE);
     }
 
+    @Test
+    public void testResourceArnIsNull_returnsNotFound() {
+        request.setDesiredResourceState(ResourceModel.builder()
+                .name(ACTIVITY_NAME)
+                .build());
+
+        final ProgressEvent<ResourceModel, CallbackContext> response
+                = handler.handleRequest(proxy, request, null, logger);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
+        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.NotFound);
+    }
+
+    @Test
+    public void testResourceDoesNotExist_returnsNotFound() {
+        AmazonServiceException resourceNotFoundException = new AmazonServiceException("Activity does not exist");
+        resourceNotFoundException.setStatusCode(400);
+        resourceNotFoundException.setErrorCode(Constants.ACTIVITY_DOES_NOT_EXIST_ERROR_CODE);
```

Comment:
method is named `ResourceDoesNotExist`, but you're testing the Activity does not exist error code.

I think we need to cover both error codes since they both get cast into NotFound errors

Reply:
Renamed to `testActivityDoesNotExist_returnsNotFound`. This is the exception thrown by DescribeActivity if the activity does not exist. DeleteActivity does not throw an exception if the activity does not exist.

Reply:
The other error code (ResourceNotFound) is thrown by the TagResource/UntagResource API. I'll make sure those are tested there.